docs/performance_tuning.md

```diff
@@ -0,0 +1,285 @@
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Deploying your trained model using Triton
+
+Given a trained model, how do I deploy it at-scale with an optimal configuration using Triton Inference Server? 
+This document is here to help answer that.
+
+For those who like a [high level overview](#overview), below is the common flow for most use cases. 
+
+For those who wish to jump right in, skip to the [end-to-end example](#end-to-end-example).
+
+## Overview
+
+1. Is my model compatible with Triton?
+    - If your model falls under one of Triton's [supported backends](https://github.com/triton-inference-server/backend), then we can simply try to deploy the model as described in the [Quickstart](https://github.com/triton-inference-server/server/blob/main/docs/quickstart.md) guide. 
+    For the ONNXRuntime, TensorFlow SavedModel, and TensorRT backends, the minimal model configuration can be inferred from the model using Triton's [AutoComplete](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#auto-generated-model-configuration) feature. 
+    This means that a `config.pbtxt` may still be provided, but is not required unless you want to explicitly set certain parameters. 
+    Additionally, by enabling verbose logging via `--log-verbose=1`, you can see the complete config that Triton sees internally in the server log output. 
+    For other backends, refer to the [Minimal Model Configuration](https://github.com/triton-inference-server/server/blob/main/docs/model_configuration.md#minimal-model-configuration) required to get started.
+    - If your model does not come from a supported backend, you can look into the [Python Backend](https://github.com/triton-inference-server/python_backend) or writing a [Custom C++ Backend](https://github.com/triton-inference-server/backend/blob/main/examples/README.md) to support your model. 
+    The Python Backend provides a simple interface to execute requests through a generic python script, but may not be as performant as a Custom C++ Backend. 
+    Depending on your use case, the Python Backend performance may be a sufficient tradeoff for the simplicity of implementation.
+
+2. Can I run inference on my served model?
+    - Assuming you were able to load your model on Triton, the next step is to verify that we can run inference requests and get a baseline performance benchmark of your model. 
+    Triton's [Perf Analyzer](https://github.com/triton-inference-server/server/blob/main/docs/perf_analyzer.md) tool specifically fits this purpose. 
+    Here is a simplified output for demonstration purposes:
+
+    ```
+    # NOTE: "my_model" represents the model currently being served by 
```

Comment:
What is the 'previous step' in this context?

Reply:
You're right, it's a bit confusing here - I'll remove that.